### PR TITLE
Refactor Check and Sync Fork Workflow

### PR DESCRIPTION
**Pull request by itschrishudson**
The previous logic between the check workflow and sync workflow was sub-optimal as the check job would be in a failed state which would look like an issue when it wasn't.

This refactor pulls the jobs together into a single workflow, and adds an improved conditional gate, with dependency on the check job running, and then having the required boolean to trigger the sync job creating the PR.

### CODE REVIEW

File: .github/workflows/check-for-changes.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Check for Changes
+name: Check for Changes & Sync from Fork
```
Comment (mihaiplesa) [inline .github/workflows/check-for-changes.yml:1]: might want to rename this file as well, something like `sync-from-fork.yml`

Comment (mihaiplesa) [reply .github/workflows/check-for-changes.yml:1]: and `Sync from fork` would be enough as a name here

File: .github/workflows/check-for-changes.yml [hunk 2]
```diff
@@ -22,5 +24,17 @@ jobs:
       - name: show files
         run: cat updatedfiles.txt
       - name: Have files changed
-        if: steps.git-check.outputs.changes_present == 'false'
-        run: exit 1
\ No newline at end of file
+        if: steps.checkForChanges.outputs.changes_present == 'false'
+        run: echo "No changes needed to be synchronised"
+  sync:
+    needs: checkForChanges
```
Comment (mihaiplesa) [inline .github/workflows/check-for-changes.yml:30]: what's the actual standard for naming targets? camel casing just looks a bit surprising in the logs

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/workflows/check-for-changes.yml (hunk 2 above)
Comment (itschrishudson) [reply .github/workflows/check-for-changes.yml:30]: No standard as far as I'm aware, examples appear mixed. Would you prefer we used hyphens or underscores instead?

Comment (mihaiplesa) [reply .github/workflows/check-for-changes.yml:30]: `-`

File: .github/workflows/check-for-changes.yml (hunk 1 above)
Comment (itschrishudson) [reply .github/workflows/check-for-changes.yml:1]: Fixed

File: .github/workflows/sync-from-fork.yml [hunk 3]
```diff
@@ -18,9 +20,21 @@ jobs:
...
+        run: echo ::set-output name=changes-present::$(if grep -qE "${FILES_TO_CHECK}" updatedfiles.txt; then echo 'true'; else echo 'false'; fi)
       - name: show files
         run: cat updatedfiles.txt
       - name: Have files changed
-        if: steps.git-check.outputs.changes_present == 'false'
-        run: exit 1
\ No newline at end of file
+        if: steps.check-for-changes.outputs.changes-present == 'false'
+        run: echo "No changes needed to be synchronised"
+  sync:
```
Comment (mihaiplesa) [inline .github/workflows/sync-from-fork.yml:29]: `sync-from-fork`

File: .github/workflows/sync-from-fork.yml [hunk 4]
```diff
@@ -18,9 +20,21 @@ jobs:
           git fetch upstream && git diff --name-only --diff-filter AM HEAD..remotes/upstream/master > updatedfiles.txt
       - name: Check for modified files
         id: git-check
-        run: echo ::set-output name=changes_present::$(if grep -qE "${FILES_TO_CHECK}" updatedfiles.txt; then echo 'true'; else echo 'false'; fi)
+        run: echo ::set-output name=changes-present::$(if grep -qE "${FILES_TO_CHECK}" updatedfiles.txt; then echo 'true'; else echo 'false'; fi)
       - name: show files
```
Comment (mihaiplesa) [inline .github/workflows/sync-from-fork.yml:24]: `Show files`

File: .github/workflows/sync-from-fork.yml [hunk 5]
```diff
@@ -1,15 +1,17 @@
-name: Check for Changes
+name: Sync from Fork
```
Comment (mihaiplesa) [inline .github/workflows/sync-from-fork.yml:1]: lowercase f? at least based on casing of `name` items below. or how does GitHub do for their actions?

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/workflows/sync-from-fork.yml [hunk 6]
```diff
@@ -18,9 +20,21 @@ jobs:
...
+    if: ${{ needs.check-for-changes.outputs.changes == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: tgymnich/fork-sync@v1.6.3
+        with:
+          owner: gorhill
+          base: master
+          head: master
+          auto_approve: false
+          ignore_fail: true
```
Comment (mihaiplesa) [inline .github/workflows/sync-from-fork.yml:40]: to add newline

File: .github/workflows/sync-from-fork.yml (hunk 3 above)
Comment (itschrishudson) [reply .github/workflows/sync-from-fork.yml:29]: The reference is to the job name, which in this case is 
```
jobs:
  check-for-changes:
```
The name in line one is just a friendly name for whole workflow, and the filename is distinct. No changes needed IMO.

File: .github/workflows/sync-from-fork.yml (hunk 5 above)
Comment (itschrishudson) [reply .github/workflows/sync-from-fork.yml]: fixed

File: .github/workflows/sync-from-fork.yml (hunk 6 above)
Comment (itschrishudson) [reply .github/workflows/sync-from-fork.yml]: Fixed

File: .github/workflows/sync-from-fork.yml (hunk 4 above)
Comment (itschrishudson) [reply .github/workflows/sync-from-fork.yml]: Fixed

**Review by mihaiplesa — APPROVED** [review]

**Pull request merged**